sklearn/calibration.py

```diff
@@ -543,15 +543,17 @@ def calibration_curve(y_true, y_prob, normalize=False, n_bins=5):
         onto 0 and the largest one onto 1.
 
     n_bins : int
-        Number of bins. A bigger number requires more data.
+        Number of bins. A bigger number requires more data. Bins with no data
+        points (i.e. without corresponding values in y_prob) will not be
+        returned, thus there may be fewer than n_bins in the return value.
 
     Returns
     -------
-    prob_true : array, shape (n_bins,)
-        The true probability in each bin (fraction of positives).
+    prob_true : array, shape (n_non_empty_bins,)
```

Comment:
I wonder if this should be expressed as `(n_bins,) or smaller` to avoid confusion. or `(n_bins,)` with a comment below.

Reply:
I think  `(n_bins,) or smaller` would work well.   If we just had `(n_bins,)` on the first line, however, I'd worry that it would be easy to miss the caveat/comment below.  A footnote would be another option, but your first suggestion seems clearest to me.  Should I modify the pull request?